### DISCUSSION TITLE
How to update an already indexed document?

### DISCUSSION THREAD

**Discussion by h0lg** [mikegoatly/lifti#124] [Q&A] [answered]
Is using `AddAsync<TObject>(TObject, CancellationToken)` good for updating?

I'm using lifti to index documents that can update outside of the index - so I need to re-index them on change. I'm currently using the `FullTextIndex<TKey>.AddAsync<TObject>(TObject, CancellationToken)` method (the index builder knows the key via its `.WithObjectTokenization().WithKey()` config) for both adding and updating and it seems to work without issues. But the name `AddAsync` raises the question whether it's intended to be used like that or whether I should check for an already indexed document using `FullTextIndex<TKey>.Metadata.Contains()` and remove it before.

**Comment by mikegoatly** [comment, marked as answer]
As long as you've got the default duplicate key behaviour configured (https://mikegoatly.github.io/lifti/docs/index-construction/withduplicatekeybehavior/) then the index will manage the contains/remove/add process

**Reply by mikegoatly** [reply]
But that said, the docs for the AddAsync method could do with mentioning that...

**Reply by h0lg** [reply]
> duplicate key behaviour

I was not aware of that - thanks for the hint and quick answer!
